reporter/otlp_reporter.go

```diff
@@ -601,14 +601,15 @@ func (r *OTLPReporter) getProfile() (profile *profiles.Profile, startTS, endTS u
 						fileName = execInfo.fileName
 					}
 
-					mappingAttributes := addProfileAttributes(profile, []attrKeyValue[string]{
+					mappingAttributes := []AttrIndex{
 						// Once SemConv and its Go package is released with the new
 						// semantic convention for build_id, replace these hard coded
 						// strings.
-						{key: "process.executable.build_id.gnu", value: execInfo.gnuBuildID},
-						{key: "process.executable.build_id.htlhash",
-							value: traceInfo.files[i].StringNoQuotes()},
-					}, attributeMap)
+						attrMgr.AddStringAttr("process.executable.build_id.gnu",
```

Comment:
nit: Rename `AddStringAttr` to `AddString` to avoid repetition of "Attr". Same for `AddIntAttr`.

Reply:
Done, though name additionally changed further to `AppendOptionalString` because I switched to an append-based approach to support optional attributes.